### PR TITLE
bugfix(ticdc): abort the changefeed creation if the downstream is MQ with the syncpoint feature enabled

### LINKED ISSUE #7637: sycn point can not works kafka, error message is inaccurate.
### What did you do?

create kafka changefeed, with syncpoint enabled.

### What did you expect to see?

changefeed should not be created.

### What did you see instead?

changefeed created, and report `kafka` as a invalid scheme.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by charleszheng44**
### What problem does this PR solve?
Issue Number: close #7637 

### What is changed and how it works?
bugfix(ticdc): abort the changefeed creation if the downstream is MQ with the syncpoint feature enabled

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

#### Questions 
No

##### Will it cause performance regression or break compatibility?
No

##### Do you need to update user documentation, design documentation or monitoring documentation?
No

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -335,6 +336,19 @@ func (o *ownerImpl) ValidateChangefeed(info *model.ChangeFeedInfo) error {
 	if err != nil {
 		return cerror.ErrAPIInvalidParam.GenWithStack("invalid sink URI %s", err)
 	}
+	// kafka changefeed does not support the syncpoint feature
+	switch sinkURI.Scheme {
```
[the remark this answers is not in the document]
Comment (charleszheng44) [reply cdc/owner/owner.go]: fixed

File: cdc/sink/validator/validator.go [hunk 2]
```diff
@@ -36,6 +36,15 @@ func Validate(ctx context.Context, sinkURI string, cfg *config.ReplicaConfig) er
 		return err
 	}
 
+	// syncpoint feature is only available for MySQL downstream
+	if cfg.EnableSyncPoint && !sink.IsMySQLCompatibleScheme(uri.Scheme) {
```
[the remark this answers is not in the document]
Comment (charleszheng44) [reply cdc/sink/validator/validator.go:40]: I am not sure.  There is no existing field in the `ReplicaConifg` struct, which can indicate the sink scheme.

[the remark this answers is not in the document]
Comment (charleszheng44) [reply cdc/sink/validator/validator.go:40]: moved verification logic to an independent function

**Review by asddongmen — APPROVED** [review]

**Pull request merged**